### PR TITLE
Add the paho-mqtt-c library as required by the syslog-ng documentation

### PR DESCRIPTION
**Pull request by mmattel**
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [x] I have read the [contributing](https://github.com/linuxserver/docker-syslog-ng/blob/main/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description:
Referencing: [Prerequisites to using the mqtt() destination](https://www.syslog-ng.com/technical-documents/doc/syslog-ng-open-source-edition/3.37/administration-guide/21) from the syslog-ng documentation.

## Benefits of this PR and context:

By adding the required library referenced by syslog-ng documentation, no additional container tweaking is neccessary.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -30,7 +30,8 @@ RUN \
     syslog-ng-stomp \
     syslog-ng-tags-parser \
     syslog-ng-xml \
-    py3-syslog-ng && \
+    py3-syslog-ng \
+    paho-mqtt-c && \
```
Comment (Roxedus) [inline Dockerfile]: Please sort these alphabetically

**Review by Roxedus — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (mmattel) [reply Dockerfile]: fixed

Comment (aptalca) [reply Dockerfile]: P is before S

Comment (mmattel) [reply Dockerfile]: Fixed, but just to let you know that py3-syslog-ng was already at the end and not on top. means someone (tm) missed sorting before and nobody took care...

**Comment by mmattel** [comment]
The sort order is fixed now though it was not correct before this PR...
Note that I have double checked if there is a syslog-ng package possibly missing related to mqtt.
The command used was :`apk list -a | grep syslog-ng` and had no mqtt related results.
Changing this PR into review state.

@Roxedus could you re-review?

**Review by thespad — APPROVED** [review]

**Pull request merged**